src/app/modules/ontology-exploration/ontology-search/ontology-search.component.spec.ts

```diff
@@ -1,25 +1,83 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { MatAutocompleteSelectedEvent } from '@angular/material/autocomplete';
+import { of } from 'rxjs';
+import { RecursivePartial, Shallow } from 'shallow-render';
 
+import { OntologyNode } from '../../../core/models/ontology-node';
+import { OntologySearchService, SearchResult } from '../../../core/services/ontology-search/ontology-search.service';
 import { OntologySearchComponent } from './ontology-search.component';
+import { OntologySearchModule } from './ontology-search.module';
+
+
+function fromPartial<T>(partial: RecursivePartial<T>): T {
+  return partial as T;
+}
+
 
 describe('OntologySearchComponent', () => {
-  // let component: OntologySearchComponent;
-  // let fixture: ComponentFixture<OntologySearchComponent>;
-
-  // beforeEach(async(() => {
-  //   TestBed.configureTestingModule({
-  //     declarations: [ OntologySearchComponent ]
-  //   })
-  //   .compileComponents();
-  // }));
-
-  // beforeEach(() => {
-  //   fixture = TestBed.createComponent(OntologySearchComponent);
-  //   component = fixture.componentInstance;
-  //   fixture.detectChanges();
-  // });
-
-  // it('should create', () => {
-  //   expect(component).toBeTruthy();
-  // });
+  let shallow: Shallow<OntologySearchComponent>;
+  let mockSearchService: jasmine.SpyObj<OntologySearchService>;
+
+  beforeEach(() => {
+    mockSearchService = jasmine.createSpyObj<OntologySearchService>(['loadOntology', 'filter']);
+    mockSearchService.filter.and.returnValue(of([]));
+
+    shallow = new Shallow(OntologySearchComponent, OntologySearchModule)
+      .mock(OntologySearchService, mockSearchService);
+  });
+
+  it('should display the label in correct format', async () => {
+    const option = {
+      displayLabel: ['first', 'second']
+    } as SearchResult;
+    const { instance } = await shallow.render();
+
+    const value = instance.displayFormatter(option);
+    expect(value).toBe('firstsecond');
+  });
+
+  it('should get lower case string to sort in lexical order', async () => {
+    const entry = fromPartial<SearchResult>({ node: { label: 'My Label' } });
+    const { instance } = await shallow.render();
+
+    const value = instance.sortLexically(entry);
+    expect(value).toBe('my label');
+  });
+
+  it('should sort by synonym result1', async () => {
+    const entry = fromPartial<SearchResult>({ displayLabel: ['first', '(second)'] });
+    const { instance } = await shallow.render();
+
+    const value = instance.sortBySynonymResult(entry);
+    expect(value).toBe(1);
+  });
+
+  it('should sort by synonym result2', async () => {
+    const entry = fromPartial<SearchResult>({ displayLabel: ['first', 'second'] });
+    const { instance } = await shallow.render();
+
+    const value = instance.sortBySynonymResult(entry);
+    expect(value).toBe(-1);
+  });
+
+  it('should call on select when node is selected', async () => {
```

Comment:
This doesn't test anything. You replace a method with a spy and then call that method (i.e. the spy) which doesn't say anything about the actual method!